artiq/language/environment.py

```diff
@@ -352,6 +354,32 @@ def setattr_dataset(self, key, default=NoDefault, archive=True):
         dataset and of the attribute are the same."""
         setattr(self, key, self.get_dataset(key, default, archive))
 
+    def set_default_pipeline(self, pipeline):
```

Comment:
> And why not allow computation of `due_date`?
I could not think of a good use case for this - if you have one I am happy to add this.

Reply:
> What about making this `set_default_scheduling(priority=x, pipeline_name=y, flush=z, due_date=u)`

I don't have a strong preference here. My feeling was in a lot of use-cases one would only want to change one of {priority, pipeline, flush}, hence it would be cleaner to have separate setters.